lua/cmake/utils.lua

```diff
@@ -70,6 +70,76 @@ function utils.run(cmd, args, opts)
   return utils.last_job
 end
 
+local function create_terminal_buf(terminal_win_cmd)
+  local cur_win = vim.api.nvim_get_current_win()
+  vim.api.nvim_command(terminal_win_cmd)
+  local bufnr = vim.api.nvim_get_current_buf()
+  local win = vim.api.nvim_get_current_win()
+  vim.api.nvim_set_current_win(cur_win)
+  return bufnr, win
+end
+
+local terminal_buf
+
+function utils.run_in_terminal(cmd, args, opts)
+  local cur_buf = vim.api.nvim_get_current_buf()
+
+  -- TODO: We need to always delete??? otherwise click ZZ will hide it
+  if terminal_buf and vim.api.nvim_buf_is_valid(terminal_buf) then
+    vim.api.nvim_buf_set_option(terminal_buf, 'modified', false)
+  else
+    local terminal_win
+    terminal_buf, terminal_win = create_terminal_buf('belowright new')
+    if terminal_win then
+      vim.wo[terminal_win].number = false
+      vim.wo[terminal_win].relativenumber = false
+      vim.wo[terminal_win].signcolumn = 'no'
+    end
+    vim.api.nvim_win_set_height(terminal_win, config.quickfix_height)
+    vim.api.nvim_buf_set_name(terminal_buf, '[neovim-cmake-terminal]')
+  end
+  local ok, path = pcall(vim.api.nvim_buf_get_option, cur_buf, 'path')
+  if ok then
+    vim.api.nvim_buf_set_option(terminal_buf, 'path', path)
+  end
+  local jobid
+
+  local chan = vim.api.nvim_open_term(terminal_buf, {
+    on_input = function(_, _, _, data)
+      pcall(vim.api.nvim_chan_send, jobid, data)
+    end,
+  })
+
+  jobid = vim.fn.jobstart(cmd, {
+    args = args,
+    cwd = opts.cwd,
+    pty = true,
```

Comment:
@Shatur Do you know the corresponding option for `Job:new`? I wasn't able to enter inputs when using it, that's why I keeped jobstart

Reply:
Not aware of it.